gcloud/datastore/helpers.py

```diff
@@ -73,6 +73,47 @@ def find_true_dataset_id(dataset_id, connection):
     return returned_pb.key.partition_id.dataset_id
 
 
+def _get_meaning(value_pb, is_list=False):
+    """Get the meaning from a protobuf value.
+
+    :type value_pb: :class:`gcloud.datastore._entity_pb2.Value`
+    :param value_pb: The protobuf value to be checked for an
+                     associated meaning.
+
+    :type is_list: bool
+    :param is_list: Boolean indicating if the ``value_pb`` contains
+                    a list value.
+
+    :rtype: int
+    :returns: The meaning for the ``value_pb`` if one is set, else
+              :data:`None`.
+    :raises: :class:`ValueError <exceptions.ValueError>` if a list value
+             has disagreeing meanings (in sub-elements) or has some
+             elements with meanings and some without.
+    """
+    meaning = None
+    if is_list:
+        all_meaning = True
```

Comment:
@tseaver Using `HasField` is "necessary" to actually tell if a meaning is set or not. For example:

```python
>>> from gcloud.datastore import _entity_pb2
>>> v = _entity_pb2.Value()
>>> v._fields
{}
>>> v.HasField('meaning')
False
>>> v.meaning
0
>>> v._fields
{}
>>> v.HasField('meaning')
False
>>> v.meaning = 0
>>> v._fields
{<google.protobuf.descriptor.FieldDescriptor object at 0x7f54f824b050>: 0}
>>> v.HasField('meaning')
True
>>> v.meaning
0
```